### DISCUSSION TITLE
Ability to generate the waveform from a start to an end set by parameters/options

### DISCUSSION THREAD

**Discussion by stevegnr** [katspaugh/wavesurfer.js#3721] [Ideas]
Hello everyone,

It would be really great if it were possible to indicate a start time and an end time on the source file, to generate the waveform only on this extract.

For example, I indicate as source a 2-minute file, but also a “start” option = 10, and another “end” option = 30, and the waveform generated corresponds to the 20-second extract from my source file.

I don't think this function exists on wavesurfer.

Thanks a lot!

**Comment by katspaugh** [comment]
That would be a great feature indeed. I'd be happy to review a PR.
